packages/config-plugins/src/ios/Entitlements.ts

```diff
@@ -35,71 +34,83 @@ export function setAssociatedDomains(
   return entitlementsPlist;
 }
 
-export function getEntitlementsPath(projectRoot: string): string {
-  const paths = Paths.getAllEntitlementsPaths(projectRoot);
-  let targetPath: string | null = null;
+export function getEntitlementsPath(
+  projectRoot: string,
+  {
+    targetName,
+    buildConfiguration = 'Release',
+  }: { targetName?: string; buildConfiguration?: string } = {}
+): string | null {
+  const project = getPbxproj(projectRoot);
+  const xcBuildConfiguration = getXCBuildConfigurationFromPbxproj(project, {
+    targetName,
+    buildConfiguration,
+  });
+  if (!xcBuildConfiguration) {
+    return null;
+  }
+  const entitlementsPath = getEntitlementsPathFromBuildConfiguration(
+    projectRoot,
+    xcBuildConfiguration
+  );
+  return entitlementsPath && fs.existsSync(entitlementsPath) ? entitlementsPath : null;
+}
 
-  /**
-   * Add file to pbxproj under CODE_SIGN_ENTITLEMENTS
-   */
+export function getDefaultEntitlementsPath(projectRoot: string) {
   const project = getPbxproj(projectRoot);
   const projectName = getProjectName(projectRoot);
   const productName = getProductName(project);
+  return path.normalize(path.join(projectRoot, 'ios', projectName, `${productName}.entitlements`));
+}
 
-  // Use posix formatted path, even on Windows
-  const entitlementsRelativePath = slash(path.join(projectName, `${productName}.entitlements`));
-  const entitlementsPath = slash(
-    path.normalize(path.join(projectRoot, 'ios', entitlementsRelativePath))
-  );
-
-  const pathsToDelete: string[] = [];
-
-  while (paths.length) {
-    const last = slash(path.normalize(paths.pop()!));
-    if (last !== entitlementsPath) {
-      pathsToDelete.push(last);
-    } else {
-      targetPath = last;
-    }
+function getEntitlementsPathFromBuildConfiguration(
+  projectRoot: string,
+  xcBuildConfiguration: XCBuildConfiguration
+): string | null {
+  const entitlementsPathRaw = xcBuildConfiguration?.buildSettings?.CODE_SIGN_ENTITLEMENTS as
+    | string
+    | undefined;
+  if (entitlementsPathRaw) {
+    return path.normalize(path.join(projectRoot, 'ios', trimQuotes(entitlementsPathRaw)));
+  } else {
+    return null;
   }
+}
 
-  // Create a new entitlements file
-  if (!targetPath) {
-    targetPath = entitlementsPath;
-
-    // Use the default template
-    let template = ENTITLEMENTS_TEMPLATE;
+export function ensureApplicationTargetEntitlementsFileConfigured(projectRoot: string): void {
+  const project = getPbxproj(projectRoot);
+  const projectName = getProjectName(projectRoot);
+  const productName = getProductName(project);
 
-    // If an old entitlements file exists, copy it's contents into the new file.
-    if (pathsToDelete.length) {
-      // Get the last entitlements file and use it as the template
-      const last = pathsToDelete[pathsToDelete.length - 1]!;
-      template = fs.readFileSync(last, 'utf8');
+  const [, nativeTarget] = findFirstNativeTarget(project);
```

Comment:
```suggestion
  const [, applicationTarget] = findFirstNativeTarget(project);
```
I'd also change the name for `findFirstNativeTarget` to sth like `findApplicationTarget` but I guess we can't do it because of libs that depend on this?